Feat/PNL and stats on landing page

jamesondh opened the pull request:
Closes #10

jamesondh committed:
feat: total unrealized PNL on landing page (WIP)

jamesondh committed:
display/format Unrealized PNL in HomeHeader

jamesondh committed:
display total spent, total outcome tokens on landing

jamesondh committed:
landing page balances overview: use first collateralTokenMetadata instead of hardcoded token

jamesondh committed:
move balance summary label text to translations.eng.json

jamesondh commented:
![Screenshot from 2021-05-13 13-35-39](https://user-images.githubusercontent.com/4026659/118164667-56390600-b3f1-11eb-959a-678b9c8891a8.png)

FranklinWaller commented:
Looks sweet! Only the alignment for mobile is kinda weird on the right side. Maybe add a breakpoint that aligns it back to the left

FranklinWaller commented on src/services/AccountService.ts:
```diff
@@ -107,6 +116,37 @@ export async function getAccountBalancesInfo(accountId: string): Promise<Account
 
+export async function getAccountBalancesSummary(accountId: string): Promise<AccountBalancesSummary> {
+    let accountBalancesInfo = await getAccountBalancesInfo(accountId);
+    console.log(accountBalancesInfo);
```
Console.log

FranklinWaller commented on src/services/AccountService.ts:
```diff
@@ -107,6 +116,37 @@ export async function getAccountBalancesInfo(accountId: string): Promise<Account
+    const totalSpent = String(spent);
+    const outcomeTokenBalance = String(outcomeTokens);
+    // assume all collateral tokens are the same as the first for master balance
+    // TODO: account for multiple collateral tokens in total balance??
```
Yeah, that would be better. If you use the TokenViewModel you can get the price of the collateral. This would allow you to calculate all the balances in dollars.

FranklinWaller commented on src/services/AccountService.ts:
```diff
@@ -107,6 +116,37 @@ export async function getAccountBalancesInfo(accountId: string): Promise<Account
+    const outcomeTokenBalance = String(outcomeTokens);
+    // assume all collateral tokens are the same as the first for master balance
+    // TODO: account for multiple collateral tokens in total balance??
+    const collateralTokenMetadata = accountBalancesInfo.marketBalances[0].collateralTokenMetadata;
```
This would break if there are no market balances.

jamesondh committed:
fix: text alignment breakpoint for landing stats

- also remove old console.log

jamesondh committed:
use TokenViewModel instead of TokenMetadata for master collateral token balance

- add dollar amount next to total spent

jamesondh committed:
check for multiple collateral tokens; display USD value

- display collateral token balance if that's the only token, otherwise just show dollar value
- return early if no market balances

jamesondh committed:
fix: total spent calculation; misc cleanup

- remove total outcome tokens
- store collateral tokens together in array

jamesondh committed:
remove need to store collateralTokens for landing page stats

jamesondh committed:
HomeHeader: calculate PnL price using spent and PnL

jamesondh committed:
subtract total spent from PnL price calculation

jamesondh commented:
![Screenshot from 2021-05-14 21-23-07](https://user-images.githubusercontent.com/4026659/118344114-ad76cd80-b4fa-11eb-93f0-ec2d55f5548d.png)
Total spent is now shown as converted price (tested with both nDAI and wNEAR) while unrealized PnL now shows the calculated profit price too. I removed total outcome tokens because this info isn't really helpful on its own. Also added left alignment breakpoint for mobile.

jamesondh committed:
fix: use price symbol of first market balance collateral token

FranklinWaller commented on src/containers/HomeHeader/HomeHeader.tsx:
```diff
@@ -55,6 +47,14 @@ export default function HomeHeader({
+                  <>
+                      <label>{trans('home.title.summary.pnl')}</label>
+                      <p className={unrealizedPnl.gt("0") ? s.link__green : s.link__red }>
+                          {unrealizedPnl.toString()}% / {totalSpent.charAt(0)}{ (Number(totalSpent.slice(1)) + ((Number(unrealizedPnl)/100) * Number(totalSpent.slice(1)))).toFixed(2) }
```
I don't wanna nitpick (sorry) But it's better to create an object separating the value and the symbol and store that in redux. It took me a bit to understand what was going on here.

FranklinWaller commented on src/containers/HomeHeader/HomeHeader.tsx:
```diff
@@ -1,35 +1,61 @@
+                  <>
+                      <label>{trans('home.title.summary.pnl')}</label>
+                      <p className={unrealizedPnl.gt("0") ? s.link__green : s.link__red }>
+                          {unrealizedPnl.toString()}% / {totalSpent.charAt(0)}{ ((Number(unrealizedPnl)/100) * Number(totalSpent.slice(1))).toFixed(2) }
```
I don't wanna nitpick (sorry) But it is better to store the price symbol and price in an object and store that in redux instead of doing charAt & slice(). Took me a bit to understand what was happening here

jamesondh replied:
Not a nitpick, agreed that storing the priceSymbol separately is cleaner, fixed! :smile:

FranklinWaller commented:
It looks really sick @jamesondh!

jamesondh committed:
store priceSymbol in account.accountSummary

FranklinWaller commented:
LGTM
